prow/inrepconfig.md

```diff
@@ -0,0 +1,66 @@
+# Inrepoconfig
+
+Inrepoconfig is a Prow feature that allows versioning Presubmit jobs in the same repository
+that also holds the code. If enabled, Prow will use both the centrally-defined jobs and the
+ones defined in the code repository. The latter ones are dynamically loaded on-demand.
+
+To enable it, add the following to your Prows `config.yaml`:
+
+```
+in_repo_config:
+  enabled:
+    # The key can be one of "*" for "globally", "org" for the GitHub organization or "org/repo" for
+    # a specific repo. The narrowest match is used.
+    "kubernetes/kubernetes": true
+```
+
+Afterwards, you need to add a config verification job to make sure people people get told about
+mistakes in their `.prow.yaml` rather than the PR being stuck. It makes sense to define this
+job in the central repository rather than the code repository, so the `checkconfig` version used
+stays in sync with the Prow version used. It looks like this:
+
+```
+presubmits:
+  kubernetes/kubernetes:
+  - name: pull-kubernetes-validate-prow-yaml
+    always_run: true
+    decorate: true
+    # Omit when using a publicly accessible repo
+    clone_uri: git@github.com:kubernetes/kubernetes.git
+    extra_refs:
+    - org: kubernetes
+      repo: test-infra
+      base_ref: master
+      # Omit when using a publicly accessible repo
+      clone_uri: git@github.com:kubernetes/test-infra.git
+     spec:
+       containers:
+       - image: gcr.io/k8s-prow/checkconfig:v20191205-050b151d0
+         command:
+         - /app/prow/cmd/checkconfig/app.binary
+         args:
+         - --plugin-config=/home/prow/go/src/github.com/kubernetes/test-infra/path/to/plugins.yaml
```

Comment:
nit: your working directory is the root of your main clone in a GOPATH-style env so you can make this clearer by using a relative path